lib/puppet/provider/sshkey/parsed.rb

```diff
@@ -36,5 +27,25 @@
   def self.default_mode
     0644
   end
+
+  def self.default_target
```

Comment:
@joshcooper turns out this was the solution to our testing problem. I did some extra verification that this is doing what we want by looking at what's happening in `Puppet::Provider::ParsedFile.default_targets` (https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/parsedfile.rb#L350-#L369).

On OSX 10.11 with an added `Puppet.notice` to show targets in `ParsedFile`:
```
km7yss53kqhd9nq:~ root# puppet apply -e "sshkey { 'mykey': type => 'ssh-rsa', key => 'test' }"
Notice: Compiled catalog for km7yss53kqhd9nq.delivery.puppetlabs.net in environment production in 0.47 seconds
Notice: TARGETS: ["/etc/ssh/ssh_known_hosts"]
```

And in OSX 10.10:
```
kx9buipxi3om7x7:~ root# puppet apply -e "sshkey { 'mykey2': type => 'ssh-rsa', key => 'test' }"
Notice: Compiled catalog for kx9buipxi3om7x7.delivery.puppetlabs.net in environment production in 0.45 seconds
Notice: TARGETS: ["/etc/ssh_known_hosts"]

```

And lastly, with a `:target =>` specified in OSX 10.10:

```
kx9buipxi3om7x7:~ root# puppet apply -e "sshkey { 'mykey2': type => 'ssh-rsa', key => 'wootles', target => '/etc/derpface' }"
Notice: Compiled catalog for kx9buipxi3om7x7.delivery.puppetlabs.net in environment production in 0.45 seconds
Notice: FINAL: ["/etc/ssh_known_hosts", "/etc/derpface"]

```

Also verified we're still using `/etc/ssh/ssh_known_hosts` on other Linuxes.